sync/src/chain.rs

```diff
@@ -2082,16 +2082,12 @@ impl ChainSync {
 
 		if !enacted.is_empty() {
 			// Select random peers to re-broadcast transactions to.
-			let mut random = random::new();
-			let len = self.peers.len();
-			let peers = random.gen_range(0, min(len, 3));
-			trace!(target: "sync", "Re-broadcasting transactions to {} random peers.", peers);
-
-			for _ in 0..peers {
-				let peer = random.gen_range(0, len);
-				self.peers.values_mut().nth(peer).map(|mut peer_info| {
+			let peers: Vec<PeerId> = ChainSync::select_random_peers(&self.peers.keys().cloned().collect::<Vec<_>>());
+			trace!(target: "sync", "Re-broadcasting transactions to random peers.");
+			for peer in peers.iter().take(3) {
```

Comment:
Previous solution was sending to 0-2 peers after each block, current is sending to 3 peers after each block.

I think it would be better to keep average number the same, and perhaps to send only to one peer on each block (we can also get rid of unnecessary allocation in such case).

```
if !is_syncing && !enacted.is_empty() && !self.peers.is_empty() {
 let peer = random::new().get_rng(0, self.peers.len());
 ...
}